e2e/setup.go

```diff
@@ -37,6 +40,66 @@ const (
 	holeskyDA  = "disperser-holesky.eigenda.xyz:443"
 )
 
+var (
+	// set by startMinioContainer
+	minioEndpoint = ""
+	// set by startRedisContainer
+	redisEndpoint = ""
+)
+
+func init() {
+	err := startMinioContainer()
+	if err != nil {
+		panic(err)
+	}
+	err = startRedisContainer()
+	if err != nil {
+		panic(err)
+	}
+}
```

Comment:
knit - using an init pattern like this could make our code super hard to import - at least this library

Reply:
agree that this sucks... also it could at some point be needed to have new clean instances per test. wanted to talk with you about how to best refactor this.

I didn't see an obvious place where to start these containers. The separation between creating the config (one fct that sets all the config vars) and starting the endpoints makes it hard... because the actual endpoint of the containers are only known after they are started, so we'd need to start them in the "createConfig" function, which feels wrong... perhaps we need yet another function like `startDependencyContainers()` or something? And we can call it from the test main function?

Reply:
On another note though I'm not sure why anyone would want to import this package... do you have anything in mind or just saying in case?

Reply:
just saying as an in-case - maybe one day we try importing this into node software directly for cross testing framework integration

Reply:
Agree here, but want to move on to other things for now and worry about this when it actually becomes a problem (perhaps an obvious refactor will also surface once we have an actual need for this). Added a TODO comment: [a9ae57b](https://github.com/Layr-Labs/eigenda-proxy/pull/177/commits/a9ae57b85a3da6a845895b699fb22684f004f728)